### ISSUE TITLE
LayerNorm generates incorrect output in float16 when XLA is enabled.

### ISSUE THREAD

**Issue by SeaOfOcean** [pytorch/xla#3658]
## 🐛 Bug

LayerNorm generates incorrect output in float16 when XLA is enabled.

## To Reproduce

```
import torch
import torchacc
import numpy as np

use_acc = 1
dtype=torch.float16

if use_acc:
    import torch_xla.core.xla_model as xm
    import torch_xla.core.xla_builder as xb
    import torch_xla.core.xla_op_registry as xor
    device = xm.xla_device()
else:
    device = 'cuda:0'


bs=3
hidden = 10
data = np.arange(0, bs*hidden).reshape([bs, hidden]).astype(np.float16)
a = torch.tensor(data, device=device)
a = a.to(dtype)
norm_layer = torch.nn.LayerNorm(hidden).to(device)
norm_layer = norm_layer.to(dtype)
b = norm_layer(a)

out = b.cpu().detach().numpy()
print("{}: ========= \n{}".format('torchacc' if use_acc else 'pytorch', out))
```

## Expected behavior

torch fp16 without xla
```
[[-1.566  -1.219  -0.87   -0.5225 -0.1741  0.1741  0.5225  0.87    1.219
   1.566 ]
 [-1.566  -1.219  -0.87   -0.5225 -0.1741  0.1741  0.5225  0.87    1.219
   1.566 ]
 [-1.566  -1.219  -0.87   -0.5225 -0.1741  0.1741  0.5225  0.87    1.219
   1.566 ]]
```

torch fp16 with xla
```
[[-1.5664062  -1.21875    -0.8701172  -0.52246094 -0.17407227  0.17407227
   0.52246094  0.8701172   1.21875     1.5664062 ]
 [-1.5791016  -1.2275391  -0.8769531  -0.5263672  -0.17541504  0.17541504
   0.5263672   0.8769531   1.2275391   1.5791016 ]
 [-1.5439453  -1.2001953  -0.8574219  -0.51464844 -0.17150879  0.17150879
   0.51464844  0.8574219   1.2001953   1.5439453 ]]
```
the first sample seems correct with xla, while the second and third sample generates wrong outputs.

## Environment

 - Reproducible on XLA backend [GPU]:
 - torch_xla version: v1.10.0

**Comment by JackCaoG** [comment]
Hmm, this might be a bit tricky. The result still seems somewhat close. Not sure if the accuracy gap coming from XLA:GPU or the way we lower the `LayerNorm`

**Comment by JackCaoG** [comment]
I guess the todo item here is to dump the HLO of `b` and file an issue to the XLA:GPU team.

**Comment by JackCaoG** [comment]
HLO dump I get is 
```
HloModule IrToHlo.44

ENTRY %IrToHlo.44 (p0.2: f32[10], p1.24: f16[3,10], p2.39: f32[10]) -> (f16[3,10]) {
  %constant.4 = f16[] constant(0), metadata={op_type="prim__Constant" op_name="prim__Constant" source_file="layer_norm@functional.py" source_line=2501}
  %reshape.5 = f16[1]{0} reshape(f16[] %constant.4), metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %broadcast.6 = f16[1]{0} broadcast(f16[1]{0} %reshape.5), dimensions={0}, metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %reshape.7 = f16[] reshape(f16[1]{0} %broadcast.6), metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %broadcast.8 = f16[3]{0} broadcast(f16[] %reshape.7), dimensions={}, metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %constant.9 = f16[] constant(0), metadata={op_type="prim__Constant" op_name="prim__Constant" source_file="layer_norm@functional.py" source_line=2501}
  %reshape.10 = f16[1]{0} reshape(f16[] %constant.9), metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %broadcast.11 = f16[1]{0} broadcast(f16[1]{0} %reshape.10), dimensions={0}, metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %reshape.12 = f16[] reshape(f16[1]{0} %broadcast.11), metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %broadcast.13 = f16[3]{0} broadcast(f16[] %reshape.12), dimensions={}, metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %p1.24 = f16[3,10]{1,0} parameter(1), metadata={op_type="xla__device_data" op_name="xla__device_data" source_file="<module>@test.py" source_line=20}
  %reshape.25 = f16[1,3,10]{2,1,0} reshape(f16[3,10]{1,0} %p1.24), metadata={op_type="aten__view" op_name="aten__view" source_file="layer_norm@functional.py" source_line=2501}
  %constant.19 = f16[] constant(1), metadata={op_type="prim__Constant" op_name="prim__Constant" source_file="layer_norm@functional.py" source_line=2501}
  %reshape.20 = f16[1]{0} reshape(f16[] %constant.19), metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %broadcast.21 = f16[1]{0} broadcast(f16[1]{0} %reshape.20), dimensions={0}, metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %reshape.22 = f16[] reshape(f16[1]{0} %broadcast.21), metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %broadcast.23 = f16[3]{0} broadcast(f16[] %reshape.22), dimensions={}, metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %constant.14 = f16[] constant(0), metadata={op_type="prim__Constant" op_name="prim__Constant" source_file="layer_norm@functional.py" source_line=2501}
  %reshape.15 = f16[1]{0} reshape(f16[] %constant.14), metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %broadcast.16 = f16[1]{0} broadcast(f16[1]{0} %reshape.15), dimensions={0}, metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %reshape.17 = f16[] reshape(f16[1]{0} %broadcast.16), metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %broadcast.18 = f16[3]{0} broadcast(f16[] %reshape.17), dimensions={}, metadata={op_type="aten__expand" op_name="aten__expand" source_file="layer_norm@functional.py" source_line=2501}
  %batch-norm-training.26 = (f16[1,3,10]{2,1,0}, f16[3]{0}, f16[3]{0}) batch-norm-training(f16[1,3,10]{2,1,0} %reshape.25, f16[3]{0} %broadcast.23, f16[3]{0} %broadcast.18), epsilon=1e-05, feature_index=1, metadata={op_type="aten__native_batch_norm" op_name="aten__native_batch_norm" source_file="layer_norm@functional.py" source_line=2501}
  %get-tuple-element.28 = f16[3]{0} get-tuple-element((f16[1,3,10]{2,1,0}, f16[3]{0}, f16[3]{0}) %batch-norm-training.26), index=1, metadata={op_type="aten__native_batch_norm" op_name="aten__native_batch_norm" source_file="layer_norm@functional.py" source_line=2501}
  %get-tuple-element.29 = f16[3]{0} get-tuple-element((f16[1,3,10]{2,1,0}, f16[3]{0}, f16[3]{0}) %batch-norm-training.26), index=2, metadata={op_type="aten__native_batch_norm" op_name="aten__native_batch_norm" source_file="layer_norm@functional.py" source_line=2501}
  %constant.30 = f16[] constant(1.0014e-05), metadata={op_type="aten__native_batch_norm" op_name="aten__native_batch_norm" source_file="layer_norm@functional.py" source_line=2501}
  %broadcast.31 = f16[3]{0} broadcast(f16[] %constant.30), dimensions={}, metadata={op_type="aten__native_batch_norm" op_name="aten__native_batch_norm" source_file="layer_norm@functional.py" source_line=2501}
  %add.32 = f16[3]{0} add(f16[3]{0} %get-tuple-element.29, f16[3]{0} %broadcast.31), metadata={op_type="aten__native_batch_norm" op_name="aten__native_batch_norm" source_file="layer_norm@functional.py" source_line=2501}
  %rsqrt.33 = f16[3]{0} rsqrt(f16[3]{0} %add.32), metadata={op_type="aten__native_batch_norm" op_name="aten__native_batch_norm" source_file="layer_norm@functional.py" source_line=2501}
  %p2.39 = f32[10]{0} parameter(2), metadata={op_type="xla__device_data" op_name="xla__device_data" source_file="convert@module.py" source_line=942}
  %convert.40 = f16[10]{0} convert(f32[10]{0} %p2.39), metadata={op_type="xla__cast" op_name="xla__cast" source_file="convert@module.py" source_line=942}
  %broadcast.41 = f16[3,10]{1,0} broadcast(f16[10]{0} %convert.40), dimensions={1}, metadata={op_type="aten__add" op_name="aten__add" source_file="layer_norm@functional.py" source_line=2501}
  %get-tuple-element.27 = f16[1,3,10]{2,1,0} get-tuple-element((f16[1,3,10]{2,1,0}, f16[3]{0}, f16[3]{0}) %batch-norm-training.26), index=0, metadata={op_type="aten__native_batch_norm" op_name="aten__native_batch_norm" source_file="layer_norm@functional.py" source_line=2501}
  %reshape.34 = f16[3,10]{1,0} reshape(f16[1,3,10]{2,1,0} %get-tuple-element.27), metadata={op_type="aten__view" op_name="aten__view" source_file="layer_norm@functional.py" source_line=2501}
  %p0.2 = f32[10]{0} parameter(0), metadata={op_type="xla__device_data" op_name="xla__device_data" source_file="convert@module.py" source_line=942}
  %convert.3 = f16[10]{0} convert(f32[10]{0} %p0.2), metadata={op_type="xla__cast" op_name="xla__cast" source_file="convert@module.py" source_line=942}
  %broadcast.35 = f16[3,10]{1,0} broadcast(f16[10]{0} %convert.3), dimensions={1}, metadata={op_type="aten__mul" op_name="aten__mul" source_file="layer_norm@functional.py" source_line=2501}
  %multiply.36 = f16[3,10]{1,0} multiply(f16[3,10]{1,0} %reshape.34, f16[3,10]{1,0} %broadcast.35), metadata={op_type="aten__mul" op_name="aten__mul" source_file="layer_norm@functional.py" source_line=2501}
  %constant.1 = f16[] constant(1), metadata={op_type="prim__Constant" op_name="prim__Constant" source_file="layer_norm@functional.py" source_line=2501}
  %broadcast.37 = f16[3,10]{1,0} broadcast(f16[] %constant.1), dimensions={}, metadata={op_type="aten__mul" op_name="aten__mul" source_file="layer_norm@functional.py" source_line=2501}
  %multiply.38 = f16[3,10]{1,0} multiply(f16[3,10]{1,0} %multiply.36, f16[3,10]{1,0} %broadcast.37), metadata={op_type="aten__mul" op_name="aten__mul" source_file="layer_norm@functional.py" source_line=2501}
  %add.42 = f16[3,10]{1,0} add(f16[3,10]{1,0} %broadcast.41, f16[3,10]{1,0} %multiply.38), metadata={op_type="aten__add" op_name="aten__add" source_file="layer_norm@functional.py" source_line=2501}
  ROOT %tuple.43 = (f16[3,10]{1,0}) tuple(f16[3,10]{1,0} %add.42)
}
```
